docs/tutorial/Advanced/docker.rst

```diff
@@ -0,0 +1,240 @@
+
+.. code-block:: sh
+
+    $ sudo usermod -aG docker <user_name>
+
+``Warning``
+The docker group grants privileges equivalent to the root user.
+For details on how this impacts security in your system, see
+`Docker Daemon Attack Surface <https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface>`_.
+
+Usage notes
+===========
+
+Docker files
+````````````````
+
+The Docker files can be found under ``Open3D/util/docker/open3d-xvfb``::
+
+    - Dockerfile
+    - setup
+        - build.sh
+        - entrypoint.s
+        - headless_sample.py
+        - headless_sample.sh
+    - tools
+        - attach.sh
+        - build.sh
+        - delete.sh
+        - it.sh
+        - prune.sh
+        - run.sh
+        - stop.sh
+
+Dockerfile
+++++++++++
+
+``Dockerfile`` is the Docker script used to build the Open3D image.
+
+Tools
++++++
+
+We provide a number of Docker tools for convenience:
+
+- ``attach.sh``
+  Start the Open3D docker container and attach to it using a terminal interface.
+- ``build.sh``
+  Build the Open3D docker image.
+- ``delete.sh``
+  Delete the Open3D image.
+- ``it.sh``
+  Start the Open3D docker container and display container stdout.
+- ``prune.sh``
+  Delete hanging containers and images.
+- ``run.sh``
+  Run the Open3D container. Checkout Open3D and build.
+- ``stop.sh``
+  Stop the Open3D container.
+
+Building the Open3D Docker image will take approximately 10-15 minutes to complete.
+At the end the image will be ~1GB in size.
+
+Running the Open3D Docker container will perform the following steps:
+
+- git clone Open3D master to ``~/Open3D_docker``
+- copy the ``headless_sample.py`` and ``headless_sample.sh`` to ``~/Open3D_docker/build/lib/Tutorial/Advanced``
+- run and detach the Open3D container with the host path ``~/Open3D_docker`` mounted inside the container at ``/root/Open3D``
+- attach a terminal to the Open3D container for command line input from the host side
+
+The Open3D container is automatically removed when stopped.
+None of the Open3D files are removed as they in fact reside on the host due to the Docker bind mounting functionality.
+In order to keep the container around remove the ``-rm`` option in ``it.sh`` and/or ``run.sh``.
+
+Prunning images/containers is useful when modifying/testing a new image.
+
+VNC
+```
+A running Open3D container listens to port 5920 on the host.
+The ``it.sh``, ``run.sh`` and ``attach.sh`` scripts redirect host port 5920 to container port 5900.
+
+This allows remoting into the container using VNC to ``<host ip>:5920``.
+The default password is ``1234`` and can be changed in ``Open3D/issue_17/util/docker/open3d-xvfb/setup/entrypoint.sh``.
+Once connected you can use Open3D as usual.
+
+Headless rendering in terminal
+``````````````````````````````
+
+Sometimes it may be necessary to perform rendering as part of some script automation.
+In order to do this follow the next steps::
+
+$ cd <Open3D path>/utilities/docker/open3d-xvfb/tools
```

Comment:
what xvfb stands for? Is that distinctive nickname that represent the whole docker container? How just about using open3d-docker for simplicity?

Reply:
`Xvfb or X virtual framebuffer is a display server implementing the X11 display server protocol` [Wikipedia](https://en.wikipedia.org/wiki/Xvfb)
I used `xvfb` in the name to convey information about how this works. I reused a pattern I've seen in use for images published on [hub.docker.com](hub.docker.com). I vote against the use of `docker` in the name because people already know that is a docker image.